compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFoldDmaWaits.cpp

```diff
@@ -152,6 +149,116 @@ LogicalResult foldDmaWaits(const AMDAIE::AMDAIEDeviceModel &deviceModel,
   return success();
 }
 
+/// For each batch, combine the async tokens into a single NpuDmaWaitOp.
+LogicalResult updateBatchTokens(IRRewriter &rewriter,
+                                SmallVector<AMDAIE::NpuDmaWaitOp> &waitOps) {
+  // Skip if there are less than two DMA wait operations.
+  if (waitOps.size() < 2) return success();
+
+  SmallVector<Value> asyncTokens;
+  Operation *parentOp = waitOps[0]->getParentOp();
+  for (AMDAIE::NpuDmaWaitOp waitOp : waitOps) {
+    if (waitOp->getParentOp() != parentOp) {
+      return waitOp.emitError(
```

Comment:
Could you add a test case for this? And I think this doesn't have to fail and could just result in no rewrite if you put it in `canFoldByBatch` instead.

Reply:
New test added
https://github.com/nod-ai/iree-amd-aie/blob/fb6d4d28130f48120d73fe535daf54def6542acc/compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/fold_dma_waits.mlir#L73